plugin/hpsa/hpsa.py

```diff
@@ -1183,19 +1185,26 @@ def volume_cache_info(self, volume, flags=Client.FLAG_RSVD):
         flag_ram_ok = False
 
         (ctrl_num, array_num, ld_num) = self._cal_of_lsm_vol(volume)
-        ctrl_data = self._sacli_exec(
+        ctrl_data = next(iter(self._sacli_exec(
             ["ctrl", "slot=%s" % ctrl_num, "show", "config", "detail"]
-            ).values()[0]
+            ).values()))
 
         lsm_bats = self.batteries()
         for lsm_bat in lsm_bats:
             if lsm_bat.status == Battery.STATUS_OK:
                 flag_battery_ok = True
 
-        if 'Total Cache Size' in ctrl_data and \
-           _hp_size_to_lsm(ctrl_data['Total Cache Size']) > 0 and \
-           ctrl_data['Cache Status'] == 'OK':
-            flag_ram_ok = True
+        if 'Total Cache Size' in ctrl_data:
+            cache_size_str = ctrl_data['Total Cache Size']
+            # Since ssacli version 3.25, cache size is a number based in GiB.
+            try:
+                int(cache_size_str)
```

Comment:
Do you mean float here? In both case, we'll get a ValueError:
```
>>> int('1024 MB')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: invalid literal for int() with base 10: '1024 MB'
>>> int('1.0')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: invalid literal for int() with base 10: '1.0'
>>> 
```

Reply:
Sorry, not testing on python2. Python3 treat `int(1.0)` as works.

Reply:
Fixed.